### PR TITLE
[FEAT] Update provider to use config

### PR DESCRIPTION
**Pull request by plummer**
Please let me know if this is incorrect, however I don't think this provider was correctly using the  values, it seemed to be using a separate url string.

### CODE REVIEW

**Comment by plummer** [comment]
Added resolution for #8

**Comment by Joannis** [comment]
I agree that MongoKitten configuration should be integrated with Vapor config files. But since MongoKitten already has a structure for [MongoDB configuration here](https://github.com/OpenKitten/MongoKitten/blob/mongokitten4/Sources/MongoKitten/Server/ClientSettings.swift).

I also think [SSL needs to be supported](https://github.com/OpenKitten/MongoKitten/blob/mongokitten4/Sources/MongoKitten/Server/SSLSettings.swift) in the configuration.

And production applications usually rely on ReplicaSets and/or sharded databases, which means [supporting multiple hosts](https://github.com/OpenKitten/MongoKitten/blob/mongokitten4/Sources/MongoKitten/Server/ClientSettings.swift#L129).

Regardless, you can always create a database using the whole MongoDB/MongoKitten URL featureset using the following:

```
let database = try Database("mongodb://localhost")
```

Which can become complex with multiple hosts, etc, too.

File: Sources/MongoProvider/Provider.swift [hunk 1]
```diff
@@ -36,20 +36,60 @@ public final class Provider: Vapor.Provider {
...
             throw ConfigError.missingFile("mongo")
         }
         
-        guard let url = mongo["url"]?.string else {
-            throw ConfigError.missing(
-                key: ["url"],
-                file: "mongo",
-                desiredType: String.self
-            )
+        func buildUrl(mongo: Config) throws -> String {
```
Comment (Joannis) [inline Sources/MongoProvider/Provider.swift:44]: The config file could be mapped to [ClientSettings](https://github.com/OpenKitten/MongoKitten/blob/mongokitten4/Sources/MongoKitten/Server/ClientSettings.swift). That simplifies this function and improves compatibility with edge-case scenarios such as complex passwords with `%` and other URL reserved characters.

File: Sources/MongoProvider/Provider.swift [hunk 2]
```diff
@@ -36,20 +36,60 @@ public final class Provider: Vapor.Provider {
...
+                    : "\(user):\(password)@"
+            
+            return "mongodb://\(userPasswordComponent)\(host):\(port)/\(database)"
         }
         
         let maxConnectionsPerServer = mongo["maxConnectionsPerServer"]?.int ?? 100
         
+        let url = try buildUrl(mongo: mongo)
+        
         try self.init(url, maxConnectionsPerServer: maxConnectionsPerServer)
```
Comment (Joannis) [inline Sources/MongoProvider/Provider.swift:93]: You could initialize a `Server` using the ClientSettings and subscript it using the `database` name to get a Database instance.

**Review by Joannis — CHANGES_REQUESTED** [review]

**Comment by plummer** [comment]
Closing, a lot of edge cases here so i think it's easier to just use the full url.

**Pull request closed without merging**